packages/python/plotly/_plotly_utils/basevalidators.py

```diff
@@ -95,44 +95,21 @@ def copy_to_readonly_numpy_array(v, kind=None, force_numeric=False):
     }
 
     if isinstance(v, nw.Series):
-        if nw.dependencies.is_pandas_like_series(v_native := v.to_native()):
-            v = v_native
+        if v.dtype == nw.Datetime and v.dtype.time_zone is not None:
+            # Remove time zone so that local time is displayed
+            v = v.dt.replace_time_zone(None).to_numpy()
         else:
             v = v.to_numpy()
     elif isinstance(v, nw.DataFrame):
-        if nw.dependencies.is_pandas_like_dataframe(v_native := v.to_native()):
-            v = v_native
-        else:
-            v = v.to_numpy()
-
-    if pd and isinstance(v, (pd.Series, pd.Index)):
-        # Handle pandas Series and Index objects
-        if v.dtype.kind in numeric_kinds:
-            # Get the numeric numpy array so we use fast path below
-            v = v.values
-        elif v.dtype.kind == "M":
-            # Convert datetime Series/Index to numpy array of datetimes
-            if isinstance(v, pd.Series):
-                with warnings.catch_warnings():
-                    warnings.simplefilter("ignore", FutureWarning)
-                    # Series.dt.to_pydatetime will return Index[object]
-                    # https://github.com/pandas-dev/pandas/pull/52459
-                    v = np.array(v.dt.to_pydatetime())
-            else:
-                # DatetimeIndex
-                v = v.to_pydatetime()
-    elif pd and isinstance(v, pd.DataFrame) and len(set(v.dtypes)) == 1:
-        dtype = v.dtypes.tolist()[0]
-        if dtype.kind in numeric_kinds:
-            v = v.values
-        elif dtype.kind == "M":
-            with warnings.catch_warnings():
-                warnings.simplefilter("ignore", FutureWarning)
-                # Series.dt.to_pydatetime will return Index[object]
-                # https://github.com/pandas-dev/pandas/pull/52459
-                v = [
-                    np.array(row.dt.to_pydatetime()).tolist() for i, row in v.iterrows()
-                ]
```

Comment:
I can test tomorrow (so please ignore if you already did all the checks and tests), yet my understanding was that this code was used not only in plotly express. If that's the case, and pandas can be passed directly without being wrapped into a narwhals dataframe/series, then this code block would still be relevant or even better we should run a defensive `from_native` first?!

Reply:
ah good point! it passes the tests but it could just be that there's not enough coverage here - in which case, I think a defensive `from_native` would be good 👍